R/validate_document.R

```diff
@@ -0,0 +1,46 @@
+#' Validate Documents using Glosario Definitions
+#'
+#' @param file_path path to the file to be validated
+#' @return NULL
+#' @details
+#' In case the document is not valid the function will return a warning
+#' for each term not defined and raise a final error listing the total
+#' number of terms not matched.
+#'
+#' @export
+validate_document <- function(file_path){
+  yaml_header <- rmarkdown::yaml_front_matter(file_path)
+  yaml_defines <- yaml_header$glosario$defines
+
+  file_text <- readLines(file_path)
+
+  gdef_idx <- purrr::map_lgl(file_text, stringr::str_detect, 'gdef()')
+
+  gdef_lines <- file_text[gdef_idx]
+
+  text_slugs <- purrr::map_chr(gdef_lines, stringr::str_remove_all, "`r gdef\\(") %>%
+    purrr::map_chr(stringr::str_remove_all, "'") %>%
+    purrr::map_chr(stringr::str_remove_all, "\\)`") %>%
+    purrr::map(stringr::str_split_fixed, ", ", 2) %>%
+    purrr::map_chr(~.x[, 1])
```

Comment:
I think `unique()` needs to be called here, otherwise when you're doing the comparison between the terms listed in the YAML header and the terms found in the document you can have a mismatch if some entries are repeated.

Reply:
You might also want to explicitly add `, drop = TRUE` to make sure that you don't accidentally end up with a one-column matrix (yes, it can happen).

Reply:
@zkamvar where would you add `, drop = TRUE`?

Reply:
`~.x[, 1, drop = TRUE]`